polyglot-kotlin/src/main/kotlin/org/sonatype/maven/polyglot/kotlin/serialization/ModelScriptWriter.kt

```diff
@@ -0,0 +1,916 @@
+package org.sonatype.maven.polyglot.kotlin.serialization
+
+import org.apache.maven.execution.MavenSession
+import org.apache.maven.model.*
+import org.apache.maven.plugin.logging.Log
+import org.apache.maven.project.MavenProject
+import org.codehaus.plexus.util.xml.Xpp3Dom
+import org.sonatype.maven.polyglot.execute.ExecuteContext
+import org.sonatype.maven.polyglot.kotlin.dsl.addAll
+import org.sonatype.maven.polyglot.kotlin.dsl.addAllNonNull
+import org.sonatype.maven.polyglot.kotlin.dsl.addFirstNonNull
+import org.sonatype.maven.polyglot.kotlin.dsl.cast
+import org.sonatype.maven.polyglot.kotlin.support.anyOf
+import org.sonatype.maven.polyglot.kotlin.support.isNotNull
+import java.io.Writer
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.*
+
+/**
+ * A `ModelScriptWriter` converts a Maven project model into a Kotlin script capable of producing an identical model.
+ * Instances of this class may be configured with one of two flavors of the Kotlin Model DSL. The most basic flavor,
+ * "block", consists of simply writing callbacks on each model element and using property setters to initialize its
+ * fields. All complex model elements can be configured using this "block" flavor of the DSL. The "block" flavor,
+ * however, does little on its own to reduce the code verbosity of the original XML.
+ *
+ * ```
```

Comment:
Okay ... I understand better now how you have two .. one close to maven original pom, one closer to kotlin .. question regarding that .. can they be mixed to some degree or do you have to choose? If I understand correctly you cant mix ..

Reply:
The flavors can be mixed with no affect on functionality. Kotlin is able to infer the correct function to be called.